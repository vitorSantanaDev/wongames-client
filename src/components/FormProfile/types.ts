export type FormProfileProps = {
  username?: string
  email?: string
}
