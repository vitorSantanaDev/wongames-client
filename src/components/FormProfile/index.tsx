import Button from 'components/Button'
import Heading from 'components/Heading'
import TextField from 'components/TextField'

import { FormProfileProps } from './types'

import * as S from './styles'
import Link from 'next/link'

const FormProfile = ({ email, username }: FormProfileProps) => (
  <S.Wrapper>
    <Heading lineBottom color="black" size="small">
      My profile
    </Heading>

    <S.Form>
      <TextField
        type="text"
        name="username"
        placeholder="Username"
        label="Username"
        initialValue={username}
      />
      <TextField
        disabled
        type="email"
        name="email"
        label="E-mail"
        placeholder="E-mail"
        initialValue={email}
      />
      <S.ButtonsContainer>
        <Link href={`/forgot-password?email=${email}`} passHref>
          <Button minimal size="medium" as="a">
            Reset Password
          </Button>
        </Link>
        <Button size="large">Save</Button>
      </S.ButtonsContainer>
    </S.Form>
  </S.Wrapper>
)

export default FormProfile
