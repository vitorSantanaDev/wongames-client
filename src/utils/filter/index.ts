import { ItemProps } from 'components/ExploreSidebar/types'
import { ParsedUrlQueryInput } from 'querystring'

export type ParceArgs = {
  queryString: ParsedUrlQueryInput
  filterItems: Pick<ItemProps, 'type' | 'name'>[]
}

export const parseQueryStringToWhere = ({
  queryString,
  filterItems
}: ParceArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const object: any = {}

  Object.keys(queryString)
    .filter((item) => item !== 'sort')
    .forEach((key) => {
      const item = filterItems.find((item) => item.name === key)
      const isCheckbox = item?.type === 'checkbox'

      object[key] = !isCheckbox
        ? queryString[key]
        : { name_contains: queryString[key] }
    })

  return object
}

export const parseQueryStringToFilter = ({
  queryString,
  filterItems
}: ParceArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const object: any = {}

  Object.keys(queryString).forEach((key) => {
    const item = filterItems.find((item) => item.name === key)
    const isCheckbox = item?.type === 'checkbox'
    const isArray = Array.isArray(queryString[key])

    object[key] = !isArray && isCheckbox ? [queryString[key]] : queryString[key]
  })

  return object
}
