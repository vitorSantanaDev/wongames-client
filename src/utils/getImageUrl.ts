export const getImageUrl = (url: string | undefined): string | null => {
  if (process.env.NEXT_PUBLIC_IMAGE_HOST) {
    return `${process.env.NEXT_PUBLIC_IMAGE_HOST}${url}`
  }

  if (url) {
    return url
  }

  return null
}
