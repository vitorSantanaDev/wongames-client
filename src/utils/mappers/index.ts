import { Banner, Game, Highlight } from 'graphql/types/schema'
import formatPrice from 'utils/formatPrice'

export const bannersMapper = (banners: Banner[]) => {
  return banners.map((banner) => ({
    title: banner.title,
    img: `http://localhost:1337${banner.image?.url}`,
    buttonLabel: banner.button.label,
    buttonLink: banner.button.link,
    subtitle: banner.subtitle,
    ...(banner.ribbon && {
      ribbon: banner.ribbon?.text,
      ribbonColor: banner.ribbon!.color,
      ribbonSize: banner.ribbon!.size
    })
  }))
}

export const gamesMapper = (games: Game[]) => {
  return games.length
    ? games.map((game) => ({
        id: game.id,
        slug: game.slug,
        title: game.name,
        developer: game.developers[0]?.name,
        image: `http://localhost:1337${game.cover?.url}`,
        price: game.price
      }))
    : []
}

export const highlightMapper = (highlight: Highlight) => {
  return (
    highlight && {
      title: highlight.title,
      subtitle: highlight.subtitle,
      alignment: highlight.alignment,
      buttonLink: highlight.buttonLink,
      buttonLabel: highlight.buttonLabel,
      floatImage: `http://localhost:1337${highlight.floatImage?.url}`,
      backgroundImage: `http://localhost:1337${highlight.background?.url}`
    }
  )
}

export const cartItemsMapper = (games: Game[]) => {
  return games.map((game) => ({
    id: game.id,
    title: game.name,
    price: formatPrice(game.price),
    img: `http://localhost:1337${game.cover?.url}`
  }))
}
