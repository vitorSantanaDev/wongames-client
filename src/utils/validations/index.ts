import Joi from 'joi'
import { UserRegisterInput } from 'graphql/types/schema'

const fieldsValidations = {
  username: Joi.string().min(5).required(),
  email: Joi.string()
    .email({ tlds: { allow: false } })
    .required(),
  password: Joi.string().required(),
  confirm_password: Joi.string()
    .valid(Joi.ref('password'))
    .required()
    .messages({ 'any.only': 'confirm password does not match with password' })
}

function getFieldsErrors(objError: Joi.ValidationResult) {
  const errors: FieldErros = {}

  if (objError) {
    objError.error?.details.forEach((err) => {
      errors[err.path.join('.')] = err.message
    })
  }

  return errors
}

export function signUpValidate(values: UserRegisterInput) {
  const schema = Joi.object(fieldsValidations)

  return getFieldsErrors(schema.validate(values, { abortEarly: false }))
}

export type FieldErros = { [key: string]: string }

type SignInValues = Omit<UserRegisterInput, 'username'>

export function signInValidate(values: SignInValues) {
  const { email, password } = fieldsValidations

  const schema = Joi.object({ email, password })

  return getFieldsErrors(schema.validate(values, { abortEarly: false }))
}
