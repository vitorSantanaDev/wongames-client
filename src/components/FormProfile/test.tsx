import { render, screen } from 'utils/test-utils'
import FormProfile from '.'

describe('<FormProfile />', () => {
  it('should render the profile form', () => {
    const { container } = render(<FormProfile />)

    const heading = screen.getByRole('heading', { name: /my profile/i })
    expect(heading).toBeInTheDocument()

    const inputName = screen.getByRole('textbox', { name: /name/i })
    expect(inputName).toBeInTheDocument()

    const inputEmail = screen.getByRole('textbox', { name: /e-mail/i })
    expect(inputEmail).toBeInTheDocument()

    const saveButton = screen.getByRole('button', { name: /save/i })
    expect(saveButton).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
