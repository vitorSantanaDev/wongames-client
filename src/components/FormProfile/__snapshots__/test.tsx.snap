// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormProfile /> should render the profile form 1`] = `
.c6 {
  border: none;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FAFAFA;
  padding: 0.8rem;
  border-radius: 0.4rem;
  background: linear-gradient(180deg,#ff605f 0%,#f062c0 50%);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
  background: none;
  color: #F231A5;
}

.c6:hover {
  background: none;
}

.c6:hover {
  color: #e20e8d;
}

.c7 {
  border: none;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FAFAFA;
  padding: 0.8rem;
  border-radius: 0.4rem;
  background: linear-gradient(180deg,#ff605f 0%,#f062c0 50%);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c7:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 1.6rem;
}

.c0::after {
  content: '';
  left: 0;
  bottom: -1rem;
  width: 5rem;
  position: absolute;
  border-bottom: 0.5rem solid #F231A5;
}

.c0::after {
  width: 3rem;
}

.c2 {
  font-size: 1.4rem;
  color: #030517;
  cursor: pointer;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 0.2rem;
  background: #EAEAEA;
  padding: 0 1.6rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c3:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c4 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem;
  padding-left: 1.6rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c4:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 2.4rem #EAEAEA inset;
  -webkit-filter: none;
  filter: none;
}

.c1 {
  max-width: 100%;
  display: grid;
  gap: 1.6rem;
}

.c1 > button {
  margin-top: 5.6rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

@media (min-width:768px) {
  .c1 {
    gap: 3.2rem;
    grid-template-columns: 1fr 1fr;
  }

  .c1 > button {
    grid-column: 2;
    justify-self: end;
    margin-top: 0;
  }
}

@media (min-width:768px) {
  .c5 {
    grid-column: 2;
  }
}

<main
  class=""
>
  <h2
    class="c0"
    color="black"
  >
    My profile
  </h2>
  <form
    class="c1"
  >
    <div
      class=""
    >
      <label
        class="c2"
        for="username"
      >
        Username
      </label>
      <div
        class="c3"
      >
        <input
          class="c4"
          id="username"
          name="username"
          placeholder="Username"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class=""
      disabled=""
    >
      <label
        class="c2"
        for="email"
      >
        E-mail
      </label>
      <div
        class="c3"
      >
        <input
          class="c4"
          disabled=""
          id="email"
          name="email"
          placeholder="E-mail"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="c5"
    >
      <a
        class="c6"
        href="/forgot-password?email=undefined"
      >
        <span>
          Reset Password
        </span>
      </a>
      <button
        class="c7"
      >
        <span>
          Save
        </span>
      </button>
    </div>
  </form>
</main>
`
